conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/UndertowRuntime.java

```diff
@@ -32,4 +32,8 @@
 
     /** Provides the {@link AuthorizationExtractor} used to read auth tokens from request headers. */
     AuthorizationExtractor auth();
+
+    /** Provides the {@link SafeParamStorer} used to store safe parameters (header, query, body params) for further
+     * processing. (e.g logging) */
+    SafeParamStorer safeParamStorer();
```

Comment:
Not a huge fan of adding this here, as everything reachable from `UndertowRuntime` previously really neatly map to first-class concepts in Conjure... feels like this will set a precent that turns the UndertowRuntime class into a bit of a grab-bag of disparate functionality.

Reply:
Not really sure how this is better than just making up an Attachment key and directly calling `exchange.putAttachment` with the correctly-presized map?

Reply:
@cakofony thoughts on the above ?

I could put a key as a public static field of a new Attachments class conjure-undertow-lib.

Reply:
The attachment would have to live in conjure-undertow-lib, despite being an implementation detail, the rest of our attachments live in the runtime. Attachment keys are based on object identity, so we must share the code that defines the key rather than a stringly matching property.

> Not a huge fan of adding this here, as everything reachable from UndertowRuntime previously really neatly map to first-class concepts in Conjure.

This is a justification for not supporting safe-logging marked parameters at all, which I wouldn't particularly mind, but I think a lot of folks would! If it's something that we actually want to support, the implementation belongs somewhere in the runtime imo.

I wonder if we can map this feature to a conjure concept?

```java
// invocation generated for any parameter with markers. Would we need a better pojo to describe parameter type (body, path, etc) and other metadata (java type) here?
// marker sets would be extracted to Endpoint level constants, and would not use guava
runtime.markers().markParameter("parameterName", parameterValue, ImmutableSet.of(Safe.class));
```

> Also adding methods to an interface that other people can implement technically breaks compile :/

That's why the javadoc for `UndertowRuntime` includes this, giving us (theoretically) the freedom add methods, just not to remove them.

```java
 * The {@link UndertowRuntime} and provided interfaces {@link BodySerDe}, {@link PlainSerDe}, and
 * {@link AuthorizationExtractor} are internal API, no guarantees are made for custom implementations.
```

Reply:
@cakofony Interesting proposal.
If we go in this direction:
-  I'd like indeed to differentiate between the different type of parameters (either by passing an enum ParameterType) or by having different methods
- markers should be strings instead of class and represented as such in the IR.